src/components/NewWorkspaceModal.test.ts

```diff
@@ -154,17 +268,110 @@ describe('NewWorkspaceModal', () => {
         );
       });
 
-      const projectSelect = new SelectHelper(screen.getByLabelText('Billing project *'), user);
-      const availableBillingProjectOptions = await projectSelect.getOptions();
-      // Remove icon name from option label.
-      // The icon names are only present in tests. They're the result of a configured transform.
-      const availableBillingProjects = availableBillingProjectOptions.map((opt) => opt.split('.svg')[1]);
-
       // Assert
-      expect(availableBillingProjects).toEqual(expectedBillingProjects);
+      expect(await getAvailableBillingProjects(user)).toEqual(expectedBillingProjects);
     }
   );
 
+  it('Hides Azure billing projects when cloning a GCP workspace', async () => {
+    const user = userEvent.setup();
+    const mockAjax: DeepPartial<AjaxContract> = {
+      Workspaces: {
+        workspace: () => ({
+          checkBucketLocation: jest.fn().mockResolvedValue({
+            location: 'US-CENTRAL1',
+            locationType: 'location-type',
+          }),
+        }),
+      },
+      Billing: {
+        listProjects: async () => [gcpBillingProject, azureBillingProject],
+      },
+      ...nonBillingAjax,
+    };
+    asMockedFn(Ajax).mockImplementation(() => mockAjax as AjaxContract);
+
+    // Act
+    await act(async () => {
+      render(
+        h(NewWorkspaceModal, {
+          cloneWorkspace: defaultGoogleWorkspace,
+          onDismiss: () => {},
+          onSuccess: () => {},
+        })
+      );
+    });
+
+    // Assert
+    expect(await getAvailableBillingProjects(user)).toEqual(['Google Billing Project']);
+  });
+
+  it('Hides GCP billing projects when cloning an Azure workspace', async () => {
+    const user = userEvent.setup();
+
+    asMockedFn(Ajax).mockImplementation(
+      () =>
+        ({
+          Billing: {
+            listProjects: async () => [gcpBillingProject, azureBillingProject],
```

Comment:
Maybe add the protected data Azure billing project here as well to make sure it's not getting excluded when cloning non-protected data workspaces?